bpf/lib/nat.h

```diff
@@ -312,24 +313,25 @@ snat_v4_nat_handle_mapping(struct __ctx_buff *ctx,
 	if (*state) {
 		int ret;
 		struct ipv4_ct_tuple rtuple = {};
-
-		if (target->addr == (*state)->to_saddr &&
+		/* reverse SNAT entries could be missing due to LRU eviction */
+		set_v4_rtuple(tuple, *state, &rtuple);
+		rstate = __snat_lookup(map, &rtuple);
+		if (rstate &&
+		    target->addr == (*state)->to_saddr &&
 		    needs_ct == (*state)->common.needs_ct) {
 			barrier_data(*state);
 			return 0;
 		}
 
-		/* Recreate the SNAT and RevSNAT entries if the source IP is stale.
-		 * Otherwise, the packet will be erroneously SNATed with the stale
-		 * source IP.
+		/* Recreate the SNAT and RevSNAT entries if the source IP is stale,
```

Comment:
consider: when there are multiple cases, it reads better as a list and allows items to be added without touching other comment lines unnecessarily:

```
Recreate the SNAT and RevSNAT entries if any of:
- Source IP is stale
- RevSNAT entry is missing

Otherwise, the packet will be erroneously SNATed with the stale source IP or dropped.
```

Reply:
Done.